rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -288,11 +289,11 @@ private int readInternal(byte[] buffer, int offset, int length)
       builder.add(readService.submit(readRequestChain));
     }
 
-    List<ListenableFuture<Integer>> futures = builder.build();
-    for (ListenableFuture<Integer> future : futures) {
+    List<ListenableFuture<Long>> futures = builder.build();
+    for (ListenableFuture<Long> future : futures) {
       // exceptions handled in caller
       try {
-        sizeRead += future.get();
+        sizeRead = Math.addExact(sizeRead, Math.toIntExact(future.get()));
```

Comment:
Why is the `+=` replaced with `=` here ?

Reply:
Simple addition is replaced by `Math.addExact` given the interfaces now return long.